src/Behat/Testwork/Cli/Application.php

```diff
@@ -126,11 +126,26 @@ public function doRun(InputInterface $input, OutputInterface $output): int
             $this->configurationLoader->setConfigurationFilePath($path);
         }
 
-        $this->add($this->createCommand($input, $output));
+        $this->addCommand($this->createCommand($input, $output));
 
         return parent::doRun($input, $output);
     }
 
+    public function addCommand(callable|SymfonyCommand $command): ?SymfonyCommand
+    {
+        // Provide compatibility with all supported symfony/console versions
+        // Attempt to use the `addCommand` method added in symfony/console 7.4.0
+        // (`add` was deprecated in 7.4.0 and removed in 8.0.0)
+        if (method_exists(parent::class, 'addCommand')) {
+            return parent::addCommand($command);
+        }
+
+        // Otherwise assert we are on an older version with `add` and call that.
+        assert(method_exists($this, 'add'));
```

Comment:
This is the main functional difference to the patch in the 4.x branch - that was just using a ternary:
https://github.com/Behat/Behat/blob/a4302410cb7badc3ce95710969318e2f0c918749/src/Behat/Testwork/Cli/Application.php#L130

But of course there is no guarantee that `add` exists just because `addCommand` does not, so once 8.0 was actually released (and therefore installed in the phpstan build) it reported that the `add` method does not exist. Asserting here avoids that.